Firebase/Auth/Source/Public/FIRAuth.h

```diff
@@ -416,26 +416,57 @@ NS_SWIFT_NAME(Auth)
 
 
     @remarks See `FIRAuthErrors` for a list of error codes that are common to all API methods.
-
-    @remarks This method will only exist until the next major Firebase release following 4.x.x.
-        After the next major release the method `signInWithEmail:password:completion:` will support
-        the `FIRAuthDataResultCallback`.
  */
 - (void)signInAndRetrieveDataWithEmail:(NSString *)email
                               password:(NSString *)password
-                            completion:(nullable FIRAuthDataResultCallback)completion;
+                            completion:(nullable FIRAuthDataResultCallback)completion
+                                __attribute__((deprecated));
 
 /** @fn signInWithCredential:completion:
-    @brief Convenience method for `signInAndRetrieveDataWithCredential:completion:` This method
-        doesn't return additional identity provider data.
+    @brief Asynchronously signs in to Firebase with the given 3rd-party credentials (e.g. a Facebook
+        login Access Token, a Google ID Token/Access Token pair, etc.) and returns additional
+        identity provider data.
+
+    @param credential The credential supplied by the IdP.
+    @param completion Optionally; a block which is invoked when the sign in flow finishes, or is
+        canceled. Invoked asynchronously on the main thread in the future.
+
+    @remarks Possible error codes:
+
+        + `FIRAuthErrorCodeInvalidCredential` - Indicates the supplied credential is invalid.
```

Comment:
These should likely be in Swift style as well:

```
`.invalidCredential` - Indicates ....
`.operationNotAllowed` - Indicates that ...

etc, etc 
```

Reply:
Maybe we should leave swiftifying the API for another pull request.